timely_beliefs/utils.py

```diff
@@ -10,3 +13,113 @@ def enforce_utc(dt: datetime) -> pd.Timestamp:
             "The timely-beliefs package does not work with timezone-naive datetimes. Please localize your datetime."
         )
     return pd.Timestamp(dt.astimezone(utc))
+
+
+def all_of_type(l: Sequence, element_type) -> bool:
+    """Return true if all elements in sequence are of the same type."""
+    for item in l:
+        if type(item) != element_type:
+            return False
+    return True
+
+
+def with_error_settings(**new_settings):
+    """
+    Function decorator to apply numpy error setting only to the decorated function.
+
+    :param new_settings: see https://docs.scipy.org/doc/numpy/reference/generated/numpy.seterr.html
+    """
+
+    def decorator(fn):
+        @wraps(fn)
+        def wrapper(*args, **kwargs):
+            old_settings = np.seterr(**new_settings)
+            out = fn(*args, **kwargs)
+            np.seterr(**old_settings)
+            return out
+
+        return wrapper
+
+    return decorator
+
+
+@with_error_settings(divide="ignore")
+def divide_ignore(*args, **kwargs):
+    return np.divide(*args, **kwargs)
+
+
+def replace_multi_index_level(
```

Comment:
Something's off because it seems to me this function exists twice:

```
± grep -r "def replace_multi_index_level" timely_beliefs
timely_beliefs/beliefs/utils.py:def replace_multi_index_level(
timely_beliefs/utils.py:def replace_multi_index_level(
```